bpf-gpl/tc.c

```diff
@@ -1272,6 +1253,82 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct __sk_buff *skb,
 	}
 }
 
+static CALI_BPF_INLINE struct fwd calico_tc_skb_icmp_accepted(struct __sk_buff *skb, struct iphdr *ip, 
+						uint8_t type, uint8_t code)
+{
+	uint32_t fib_flags = 0;
+	__be32 un = 0;
+	int rc = TC_ACT_UNSPEC;
+	enum calico_reason reason = CALI_REASON_UNKNOWN;
+	uint32_t seen_mark;
+
+	if (code == ICMP_FRAG_NEEDED)
+	{
+		struct {
+			__be16  unused;
+			__be16  mtu;
+		} frag = {
+			.mtu = host_to_be16(TUNNEL_MTU),
+		};
+
+		un = *(__be32 *)&frag;
```

Comment:
we may perhaps reuse one of the 32bit fields like `pol_rc` or `tun_ip`, which are not going to be used downstream and pass the `un` from where the icmp response type is decide rather than having a special case here.